src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -50,9 +48,7 @@ public static class MSBuildLocator
         /// <remarks>
         ///     If any Microsoft.Build assemblies are already loaded into the current AppDomain, the value will be false.
         /// </remarks>
-        public static bool CanRegister => !IsRegistered && !LoadedMsBuildAssemblies.Any();
-
-        private static IEnumerable<Assembly> LoadedMsBuildAssemblies => AppDomain.CurrentDomain.GetAssemblies().Where(IsMSBuildAssembly);
+        public static bool CanRegister => !IsRegistered && !loadedAssemblies.Any();
```

Comment:
Naturally looking at the diff that's the result of the change I asked you to make revealed to me that I was wrong and this isn't what we want. I'll push a fix.